engine/access/rest/websockets/handler.go

```diff
@@ -0,0 +1,70 @@
+package websockets
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/common"
+	"github.com/onflow/flow-go/engine/access/state_stream"
+	"github.com/onflow/flow-go/engine/access/state_stream/backend"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type Handler struct {
+	*common.HttpHandler
+
+	logger          zerolog.Logger
+	websocketConfig *Config
+	streamApi       state_stream.API
+	streamConfig    backend.Config
+}
+
+var _ http.Handler = (*Handler)(nil)
+
+func NewWebSocketHandler(
+	logger zerolog.Logger,
+	config *Config,
+	chain flow.Chain,
+	streamApi state_stream.API,
+	streamConfig backend.Config,
+	maxRequestSize int64,
+) *Handler {
+	return &Handler{
+		HttpHandler:     common.NewHttpHandler(logger, chain, maxRequestSize),
+		websocketConfig: config,
+		logger:          logger,
+		streamApi:       streamApi,
+		streamConfig:    streamConfig,
+	}
+}
+func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	//TODO: change to accept topic instead of URL
+	logger := h.HttpHandler.Logger.With().Str("websocket_subscribe_url", r.URL.String()).Logger()
+
+	err := h.HttpHandler.VerifyRequest(w, r)
+	if err != nil {
+		// VerifyRequest sets the response error before returning
+		logger.Warn().Err(err).Msg("error validating websocket request")
```

Comment:
I still think we should make this a debug message. what tends to happen is we eventually get a client that spams with invalid requests, resulting in a large volume of logs. if we log these messages above info level, the operators then have to reduce their logging to avoid getting spammed.

instead, it's best to not log any errors that result from client requests as debug